File: Back_End/src/main/java/it/davidgualberto/apt/payloads/JwtResponse.java
package it.davidgualberto.apt.payloads;

import java.util.List;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class JwtResponse {
	private String token;
	private String type = "Bearer";
	private int id;
	private String username;
	private String email;
	private String name;
	private String surname;
	private String city;
	
	public JwtResponse(String accessToken, int id, String username, String email, String name, String surname, String city) {
		this.token = accessToken;
		this.id = id;
		this.username = username;
		this.email = email;
		this.name = name;
		this.surname = surname;
		this.city = city;
		}
}

File: Front_End/src/app/interfaces/jwt-response.ts
import { Reservation } from "./reservation";
import { Favorite } from "./restaurant";

export interface JwtResponse {
token: string,
type: string,
id: number,
username: string,
email:string,
name:string,
surname:string,
city:string
}

export interface User {
  id:number;
  name: string;
  surname: string;
  email: string;
  username: string;
  password: string;
  city: string;
  street: string;
  number: number;
  postCode: number;
  }

  export interface UserProfile {
    id:number;
    name: string;
    surname: string;
    email: string;
    username: string;
    password: string;
    city: string;
    street: string;
    number: number;
    postCode: number;
    reservation: Reservation[];
    favRestaurant: Favorite[];
    }
